sdk/messaging/azservicebus/receiver.go

```diff
@@ -205,6 +209,10 @@ func (r *Receiver) ReceiveMessages(ctx context.Context, maxMessages int, options
 		return nil, errors.New("receiver is already receiving messages. ReceiveMessages() cannot be called concurrently")
 	}
 
+	var err error
```

Comment:
In this case we're actually returning internal.TransformErr(err), not 'err', directly to the client. Is the span intended to match that?

(if so, we can go with the named return values trick again)

Reply:
Let me test this out -- I don't like what `internal.TransformErr` gives me because everything is just `exported.Error` and not very meaningful to the user. I do have a little workaround to unwrap the error in `endSpan` though, so maybe this will just work out

Reply:
I lied and will have to keep these in 2 places!

Reply:
I think it is what we want:

1. It's the type the user will ultimately see
2. The error message of the Error actually includes the inner-most error, so we don't lose any information there and it includes the classification code, which user's can use, if they want to process the error in code.

For instance:

```go
err := NewError(CodeConnectionLost, nil)
require.Equal(t, err.Error(), "(connlost): unknown error")

err = NewError(CodeLockLost, &amqp.Error{Condition: amqp.ErrCond("com.microsoft:message-lock-lost")})
require.Equal(t, err.Error(), "(locklost): *Error{Condition: com.microsoft:message-lock-lost, Description: , Info: map[]}")
```

Reply:
update: resolved this in offline convo and got rid of all the bad `var err error`'s